// redux-routing/lib/createMiddleware.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = createMiddleware;

var _querystring = require('querystring');

var _querystring2 = _interopRequireDefault(_querystring);

var _url = require('url');

var _url2 = _interopRequireDefault(_url);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function createMiddleware(History) {
  return function (store) {
    var history = void 0;

    if (History) {
      history = new History(store);
      history.listen();
    }

    return function (next) {
      return function (action) {
        if (!/^@@redux-routing/.test(action.type)) {
          return next(action);
        }

        var parsed = _url2.default.parse(action.href);

        var location = {
          hash: parsed.hash || undefined,
          pathname: parsed.pathname,
          search: parsed.search || undefined
        };

        var query = void 0;

        if (parsed.query) {
          query = _querystring2.default.parse(parsed.query);
        }

        var result = next(Object.assign({}, action, {
          href: _url2.default.format(location),
          location: location,
          query: query
        }));

        if (history) {
          history.update(result);
        }

        return result;
      };
    };
  };
}
