// redux-routing/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.replace = exports.navigate = exports.route = exports.reducer = exports.match = exports.History = exports.Hash = exports.createMiddleware = undefined;

var _actions = require('./actions');

Object.defineProperty(exports, 'navigate', {
  enumerable: true,
  get: function get() {
    return _actions.navigate;
  }
});
Object.defineProperty(exports, 'replace', {
  enumerable: true,
  get: function get() {
    return _actions.replace;
  }
});

var _createMiddleware = require('./createMiddleware');

var _createMiddleware2 = _interopRequireDefault(_createMiddleware);

var _Hash = require('./Hash');

var _Hash2 = _interopRequireDefault(_Hash);

var _History = require('./History');

var _History2 = _interopRequireDefault(_History);

var _match = require('./match');

var _match2 = _interopRequireDefault(_match);

var _reducer = require('./reducer');

var _reducer2 = _interopRequireDefault(_reducer);

var _route = require('./route');

var _route2 = _interopRequireDefault(_route);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.createMiddleware = _createMiddleware2.default;
exports.Hash = _Hash2.default;
exports.History = _History2.default;
exports.match = _match2.default;
exports.reducer = _reducer2.default;
exports.route = _route2.default;
